src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Export.cs

```diff
@@ -32,10 +32,11 @@ public byte[] Export(CngKeyBlobFormat format)
                 throw errorCode.ToCryptographicException();
 
             byte[] buffer = new byte[numBytesRequired];
-            errorCode = Interop.NCrypt.NCryptExportKey(_keyHandle, IntPtr.Zero, format.Format, IntPtr.Zero, buffer, numBytesRequired, out numBytesRequired, 0);
+            errorCode = Interop.NCrypt.NCryptExportKey(_keyHandle, IntPtr.Zero, format.Format, IntPtr.Zero, buffer, buffer.Length, out numBytesRequired, 0);
```

Comment:
You're trying to keep the buffer argument and the length argument as in sync as possible.  In the other call site, the numBytesNeeded is used to create the propertyValue array which is used to create the pPropertyValue pointer, so two hops between the numBytesNeeded and the pPropertyValue, whereas there's only one hop between propertyValue (and propertyValue.Length) and pPropertyValue.  Plus the names are almost identical, so there's a clear connection between them: looking just at the call site, I get the connection between "pPropertyValue" and "propertyValue.Length" much more so than I do between "pPropertyValue" and "numBytesNeeded".

If it were me and I were making this change, I'd change both.  But it's not a big deal.